rules/TypeDeclaration/Rector/ClassMethod/AddReturnTypeDeclarationBasedOnParentClassMethodRector.php

```diff
@@ -124,26 +125,28 @@ public function refactor(Node $node): ?Node
     private function getReturnTypeRecursive(ClassMethod $classMethod): ?Type
     {
         $returnType = $classMethod->getReturnType();
+        if ($returnType !== null) {
+            return $this->staticTypeMapper->mapPhpParserNodePHPStanType($returnType);
+        }
 
-        if ($returnType === null) {
-            $parentMethodReflection = $this->parentClassMethodTypeOverrideGuard->getParentClassMethod($classMethod);
-            if (! $parentMethodReflection instanceof MethodReflection) {
+        $parentMethodReflection = $this->parentClassMethodTypeOverrideGuard->getParentClassMethod($classMethod);
+        while ($parentMethodReflection instanceof MethodReflection) {
+            if ($parentMethodReflection->isPrivate()) {
                 return null;
             }
 
-            $parentClassMethod = $this->astResolver->resolveClassMethodFromMethodReflection($parentMethodReflection);
```

Comment:
`astResolver->resolveClassMethodFromMethodReflection` is super in-efficient because it needs to parse the whole file into nodes